ci-operator/step-registry/ipi/conf/gcp/aws-route53/ipi-conf-gcp-aws-route53-commands.sh

```diff
@@ -0,0 +1,136 @@
+      \"Type\": \"$recordtype\",
+      \"TTL\": 60,
+      \"ResourceRecords\": [{\"Value\": \"$record\"}]
+    }
+  }"
+  jq --argjson jd "$json_delete" '.Changes += [$jd]' "${SHARED_DIR}"/dns-delete.json > "${SHARED_DIR}"/tmp-delete.json && mv "${SHARED_DIR}"/tmp-delete.json "${SHARED_DIR}"/dns-delete.json
+}
+
+function find_out_api_and_ingress_ip_addresses() {
+  local -r infra_id="$1"
+  local -r region="$2"
+  local -r out_file="$3"
+  local api_ip_address ingress_ip_address ingress_forwarding_rule
+
+  api_ip_address=$(gcloud compute forwarding-rules describe --global "${infra_id}-apiserver" --format json | jq -r .IPAddress)
+  if [[ -z "${api_ip_address}" ]]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to find the API forwarding-rule."
+    ret=1
+  fi
+
+  ingress_forwarding_rule=$(gcloud compute target-pools list --format=json --filter="instances[]~${infra_id}" | jq -r .[].name)
+  if [[ -n "${ingress_forwarding_rule}" ]]; then
+    ingress_ip_address=$(gcloud compute forwarding-rules describe --region "${region}" "${ingress_forwarding_rule}" --format json | jq -r .IPAddress)
+  else
+    ingress_ip_address=""
+    echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to find the INGRESS forwarding-rule."
+    ret=1
+  fi
+
+  echo "$(date -u --rfc-3339=seconds) - INFO: Populate the file '${out_file}' with API server IP (${api_ip_address}) and INGRESS server IP (${ingress_ip_address})..."
+  cat > "${out_file}"  << EOF
+${api_ip_address}
+${ingress_ip_address}
+EOF
+}
+
+GCP_BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+if [[ -n "${BASE_DOMAIN}" ]]; then
+  GCP_BASE_DOMAIN="${BASE_DOMAIN}"
+fi
+
+cluster_name=${NAMESPACE}-${UNIQUE_HASH}
+cluster_domain="${cluster_name}.${GCP_BASE_DOMAIN}"
+INFRA_ID="$(jq -r .infraID ${SHARED_DIR}/metadata.json)"
+
+export AWS_SHARED_CREDENTIALS_FILE=/var/run/vault/gcp/.awscred
+export AWS_MAX_ATTEMPTS=50
+export AWS_RETRY_MODE=adaptive
+export HOME=/tmp
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+GCP_REGION="${LEASED_RESOURCE}"
+
+ret=0
+
+find_out_api_and_ingress_ip_addresses "${INFRA_ID}" "${GCP_REGION}" "${SHARED_DIR}/vips.txt"
+
+# Load array created in setup-vips:
+# 0: API
+# 1: Ingress
+declare -a vips
+mapfile -t vips < "${SHARED_DIR}"/vips.txt
+
+hosted_zone_id="$(aws route53 list-hosted-zones-by-name \
+            --query "HostedZones[? Config.PrivateZone != \`true\` && Name == \`${GCP_BASE_DOMAIN}.\`].Id" \
+            --output text)"
+echo "${hosted_zone_id}" > "${SHARED_DIR}/hosted-zone.txt"
+
+echo "$(date -u --rfc-3339=seconds) - INFO: Creating batch file to create DNS records"
```

Comment:
Maybe we can split this ref into two ref step, one is for generating hostname-IP mapping, just like what `gcp-provision-private-custom-dns` does, another is for provisioning public DNS on aws route53. Once that, we can re-use the step of provisioning public DNS on aws route53 if needed.

Reply:
How about one under `ci-operator/step-registry/gcp/provision/ppublic-custom-dns` for generating hostname-IP mapping, and another under `ci-operator/step-registry/aws/deprovision/route53/public-record-sets/` for provisioning public DNS on aws route53? Thanks!

Reply:
sounds good to me, thanks